Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -929,6 +929,28 @@ public function isArrayMultiLine($index)
     }
 
     /**
+     * Check if there is an anonymous class under given index.
+     *
+     * @param int $index
+     *
+     * @return bool
+     */
+    public function isAnonymousClass($index)
+    {
+        $token = $this[$index];
+
+        if (!$token->isClassy()) {
+            throw new \LogicException('No classy token at given index');
```

Comment:
> why throw an exception for non-classy

to be congruent to `isLambda` which throws an exception for token unequal to `T_FUNCTION`.
(I don't like the behavior. In my opinion it should return `false`.)

> but return false for trait / interface ?

to be pragmatic and allow code like this:

```php
if ($token->isClassy() && !$tokens->isAnonymousClass($index)) {
}
```
example: https://github.com/gharlan/PHP-CS-Fixer/blob/35e69bd27be0bc4c9c62b9c1efb6a663519a3c10/Symfony/CS/Fixer/Symfony/SelfAccessorFixer.php#L30